waku/v2/waku_relay/protocol.nim

```diff
@@ -29,19 +30,100 @@ logScope:
 const
   WakuRelayCodec* = "/vac/waku/relay/2.0.0"
 
-
-type WakuRelayResult*[T] = Result[T, string]
-
-type
-  PubsubRawHandler* = proc(pubsubTopic: PubsubTopic, data: seq[byte]): Future[void] {.gcsafe, raises: [Defect].}
-  SubscriptionHandler* = proc(pubsubTopic: PubsubTopic, message: WakuMessage): Future[void] {.gcsafe, raises: [Defect].}
+# see: https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#overview-of-new-parameters
+const topicParams = TopicParams(
+    topicWeight: 1,
+
+    # p1: favours peers already in the mesh
+    timeInMeshWeight: 0.01,
+    timeInMeshQuantum: 1.seconds,
+    timeInMeshCap: 10.0,
+
+    # p2: rewards fast peers
+    firstMessageDeliveriesWeight: 1.0,
+    firstMessageDeliveriesDecay: 0.5,
+    firstMessageDeliveriesCap: 10.0,
+
+    # p3: penalizes lazy peers. safe low value
+    meshMessageDeliveriesWeight: -5.0,
+    meshMessageDeliveriesDecay: 0.5,
+    meshMessageDeliveriesCap: 10,
+    meshMessageDeliveriesThreshold: 1,
+    meshMessageDeliveriesWindow: 5.milliseconds,
+    meshMessageDeliveriesActivation: 10.seconds,
+
+    # p3b: tracks history of prunes
+    meshFailurePenaltyWeight: -1.0,
+    meshFailurePenaltyDecay: 0.5,
```

Comment:
These parameters are the one I talked about that are not fit for waku, because they set an expected rate of messages per topic, and we can't estimate the actual rate of messages in waku

Reply:
mm i see. reading the docs i didnt got that impression. shall we zero both values?

> P₃b: Mesh Message Delivery Failures for a topic. This is a sticky parameter that counts the number of mesh message delivery failures. Whenever a peer is pruned with a negative score, the parameter is augmented by the rate deficit at the time of prune. This is intended to keep history of prunes so that a peer that was pruned because of underdelivery cannot quickly get re-grafted into the mesh. The parameter is mixed with negative weight.

Reply:
Yeah, 0 values will work

All of these parameters are very obscure, but this parameter is basically a value that keep increasing as time passes, and every time a peer sends a message, it's decreased by some amount
So if the rate at which it decreases is < than time passing, all of the peers in the mesh will be penalized (because of the negative weight)

That's a convoluted way to impose a certain cadence to the messages in the topic

Reply:
done 3d8ba07d5944d5374ccb981c06716db46e727518

Reply:
done dc70df97f859a4fa907db0be13bfd56d7a23dce8